File: backend/individual-assignment/src/services/payment.ts
import Invoice from "./invoice";
import PaymentMethod from "../models/PaymentMethod";
import Order from "./order";

export default class Payment {
  public paidAt: Date;
  public authorizationNumber: number;
  public amount: number;
  public invoice: Invoice;

  constructor(public order: Order, public paymentMethod: PaymentMethod) {
    this.paidAt = new Date();
    this.authorizationNumber = this.paidAt.getTime();
    this.amount = order.totalAmount;
    this.invoice = new Invoice(order);
  }
}

File: backend/individual-assignment/challenge/src/main/kotlin/dtos/Payment.kt
package org.example.dtos

import org.example.models.PaymentMethod
import org.example.models.impl.Order
import java.util.*

data class Payment(
    val order: Order,
    val paymentMethod: PaymentMethod
) {
    val paidAt = Date()
    val authorizationNumber = paidAt.time
    val amount = order.totalAmount
    val invoice = Invoice(order)
}
